### ISSUE TITLE
where can I find the instance of QUBO problem?

### ISSUE THREAD

**Issue by ntyz** [MQLib/MQLib#4]
I'm wondering where is the QUBO problems part, becacuse I only find the instance of maxcut problem in the instance_header_info.csv file, did the instance of QUBO problem have been converted to maxcut form？

**Comment by josilber2** [comment]
Yes, all QUBO instances have been converted to maxcut.

**Comment by ntyz** [comment]
Are all the max cut problem instances contained in txt file represented in the form of weight graph? And so the weights of points and edges of these weight graphs can be any value?

**Comment by josilber2** [comment]
Yes, you can read about the file format at https://github.com/MQLib/MQLib/blob/master/bin/README.md .

**Comment by wzmumu** [comment]
> Yes, you can read about the file format at https://github.com/MQLib/MQLib/blob/master/bin/README.md .

I copied two head line from `instance_header_info.csv`:

```
fname,n,m,comments
G55.zip,5000,12498,NAME: G55 || COMMENT: G-set problem instance downloaded from http://web.stanford.edu/~yyy/yyye/Gset/.
```
from which i know the information of `G55`.

My question is: do i have to visit the website in the comment to get the file? Or there is some way (e.g. command line ) in MQLib that i can get files without geting them one-by-one online?

**Comment by josilber** [comment]
Please read https://github.com/MQLib/MQLib/tree/master/scripts for helpful
scripts to download either one instance (downloadGraph.py) or all the
instances at once (downloadAllGraphs.py).

**Comment by wzmumu** [comment]
> Please read https://github.com/MQLib/MQLib/tree/master/scripts for helpful scripts to download either one instance (downloadGraph.py) or all the instances at once (downloadAllGraphs.py).

I used `downloadGraph.py` to download some instance with `boto` (installed via conda), however, failed in opening them (see the attachment as an example). Whether the file is damaged on cloud or while downloading?
[g001095.zip](https://github.com/user-attachments/files/17957450/g001095.zip)

**Comment by josilber** [comment]
I just checked with a somewhat recent version of python (python 3.12.4), and the following code worked fine for me to download a file and save it to a specified location on my computer. The unzipped version of the file seemed fine. Here I've updated to the boto3 package, since it seems that boto is no longer supported for python 3.12.

```
import boto3
from botocore import UNSIGNED
from botocore.client import Config

s3 = boto3.resource('s3', config=Config(signature_version=UNSIGNED))
obj = s3.Object("mqlibinstances", "g001095.zip")
str = obj.get()["Body"].read()
with open("g001095.zip", "wb") as fout:
    fout.write(str)
```
